.github/workflows/benchmarks.yml

```diff
@@ -1,83 +1,45 @@
 name: Benchmarking
 on:
+  pull_request:
   push:
     branches:
       - master
 
+env:
+  GCC_VERSION: 10
+
 jobs:
   benchmarks:
-    name: Gate benchmarks
-    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-20.04]
+        os: [ubuntu-22.04]
 
-    steps:
+    name: Gate benchmarks
+    runs-on: ${{ matrix.os }}
 
-      - name: Checkout PennyLane-Lightning master
-        uses: actions/checkout@v2
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.10.0
         with:
-          ref: master
-          path: main
+          access_token: ${{ github.token }}
+
+      - name: Checkout PennyLane-Lightning
+        uses: actions/checkout@v3
 
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v4
         name: Install Python
         with:
           python-version: '3.7'
 
       - name: Install dependencies
-        run: sudo apt-get -y -q install cmake gcc
-
-      - name: Get required Python packages
-        run: |
-          cd main
-          python -m pip install --upgrade pip
-          pip install -r requirements-dev.txt
-          pip install matplotlib
-
-      - name: Install lightning.qubit device (master)
-        run: |
-          cd main
-          python3 setup.py build_ext -i --define="CMAKE_CXX_COMPILER=$(which g++-10)"
-          pip install -e .
-
-      - name: Benchmark lightning master device
-        run: |
-          cd main
-          python .github/workflows/benchmarks/run_bench.py lightning.qubit lightning_master.json
+        run: sudo apt-get update && sudo apt-get -y -q install cmake gcc-10 g++-10 libopenblas-dev
 
-      - name: Benchmark default qubit device
+      - name: Build GBenchmark
         run: |
-          cd main
-          python .github/workflows/benchmarks/run_bench.py default.qubit default_qubit.json
+            cmake pennylane_lightning/src/ -BBuildGBench -DBUILD_BENCHMARKS=ON -DENABLE_BLAS=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER="$(which g++-$GCC_VERSION)"
+            cmake --build ./BuildGBench  --parallel 2
```

Comment:
We can just use ninja as the build target here `-GNinja`, and there will be no reason to specify the parallel build. This will allow us to autoscale if/when Github actions runners offer more cores.

Ninja may also need to be installed at line 35.